etebase_fastapi/collection.py

```diff
@@ -0,0 +1,225 @@
+import typing as t
+
+from asgiref.sync import sync_to_async
+from django.contrib.auth import get_user_model
+from django.core import exceptions as django_exceptions
+from django.core.files.base import ContentFile
+from django.db import transaction
+from django.db.models import Q
+from django.db.models import QuerySet
+from fastapi import APIRouter, Depends, status, Query
+from pydantic import BaseModel, Field
+
+from django_etebase import models
+from django_etebase.models import Collection, AccessLevels, CollectionMember
+from .authentication import get_authenticated_user
+from .execptions import ValidationError
+from .msgpack import MsgpackRoute, MsgpackResponse
+from .stoken_handler import filter_by_stoken_and_limit
+
+User = get_user_model()
+collection_router = APIRouter(route_class=MsgpackRoute)
+default_queryset = Collection.objects.all()
+
+
+Prefetch = t.Literal["auto", "medium"]
+PrefetchQuery = Query(default="auto")
+
+
+class ListMulti(BaseModel):
+    collectionTypes: t.List[bytes]
+
+
+class CollectionItemRevisionOut(BaseModel):
+    uid: str
+    meta: bytes
+    deleted: bool
+    chunks: t.List[t.Tuple[str, t.Optional[bytes]]]
+
+    class Config:
+        orm_mode = True
+
+    @classmethod
+    def from_orm_user(
```

Comment:
from_orm_context, not user. Maybe all of them should be called `context` to be honest. This would be consistent and let us pass all of the context needed easily even if we modify it. Context could be a simple data class.